src/content/docs/controller.md

```diff
@@ -2,4 +2,215 @@
+#[vexide::main]
+async fn main(peripherals: Peripherals) {
+    //                           (                )
+    let controller = peripherals.primary_controller;
+    //                               ^
+    //         [Get the primary controller. Note that since we moved the controller out of `peripherals`, we can no longer access it there.]
+    //                                   (                )
+    let partner_controller = peripherals.partner_controller;
+}
+```
+
+# Reading the controller's state
+
+<!-- TODO: add one of those fancy illustrations that look cool here. -->
+
+Reading the controller's state is simple through the `Controller::state` method. This method returns a `ControllerState` struct that contains the state of the controller's buttons and joysticks.
+
+You can read each button's state through the `button_x` properties and the joysticks' state through the `y_stick` properties on the `ControllerState` struct.
+
+Here's an example of how you can read the controller's state in a loop:
+
+```rs
+// @fold start
+#![no_std]
+#![no_main]
+
+use vexide::prelude::*;
+// @fold end
+#[vexide::main]
+async fn main(peripherals: Peripherals) {
+  let mut controller = peripherals.primary_controller;
+
+  loop {
+    //          (                )
+    let state = controller.state();
+
+    // Now you can use the state of the controller:
+    if state.button_a.is_pressed() {
+      // Do something repeatedly when the A button is pressed
+    } else {
+      // Do something else repeatedly when the A button is not pressed
+    }
+
+    sleep(Controller::UPDATE_INTERVAL).await;
+    // Use the state of the controller here
+  }
+}
+```
+
+Button states can be checked using:
+
+-   `button_a`/`button_b`/`button_x`/`button_y`
+-   `button_up`/`button_down`/`button_left`/`button_right`
+-   `button_l1`/`button_l2`/`button_r1`/`button_r2`
+-   `button_power` (note that while you can use this in your program, holding this button down _will_ turn off the controller, which you probably don't want to happen in a match)
+
+Joystick states can be checked using:
+
+-   `left_stick` (for the left joystick)
+-   `right_stick` (for the right joystick)
+
+## Button states
+
+To check whether a button is _currently pressed down_, you can use the `is_pressed` method on the button. This method returns `true` if the button is pressed and `false` otherwise.
+
+> Wait, but I want to know when a button was _just_ pressed!
+
+Don't worry! In VEXCode, you may have had to declare a separate variable just to keep track of the previous state of a button. In vexide, we do this for you!
+
+You can use the `is_now_pressed` method to check if a button was just pressed. This method returns `true` if the button was pressed in the last update (i.e., it was released the last time `Controller::state` was called and is now pressed) and `false` otherwise. This is quite useful for implementing actions like toggling a pneumatic piston or changing the state of a subsystem.
```

Comment:
Might be worth mentioning that if a button is pressed and unpressed in between updates the press will effectively be dropped.